util/testutil/integration/run.go

```diff
@@ -25,45 +26,32 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-type Sandbox interface {
+// Backend is the minimal interface that describes a testing backend.
+type Backend interface {
 	Address() string
-	PrintLogs(*testing.T)
+	Rootless() bool
+}
+
+type Sandbox interface {
+	Backend
+
 	Cmd(...string) *exec.Cmd
+	PrintLogs(*testing.T)
 	NewRegistry() (string, error)
-	Rootless() bool
 	Value(string) interface{} // chosen matrix value
 }
 
-type Worker interface {
-	New(...SandboxOpt) (Sandbox, func() error, error)
-	Name() string
-}
+// BackendConfig is used to configure backends created by a worker.
+type BackendConfig struct {
+	Logs map[string]*bytes.Buffer
 
-type SandboxConf struct {
 	mirror string
```

Comment:
The problem was that mirror/mv were needed to generate the options passed when making the connection. We either needed to defer making the connection to when the sandbox was created (meaning backend wouldn't have access to the address, which defeats the purpose) or we needed to pass those in earlier. If they were going to have to be passed in at the same time as the backend, it seemed pointless to have backend config and sandbox config.

We only need to pass backendconfig into sandbox because it's also carrying the mv/mirror stuff now, so it also replaces sandbox options. If this were to change it wouldn't need to be passed in and could be created in newSandbox instead.

Reply:
If we move it out of `newSandbox` we still need to set `mv` on the sandbox itself, so we'd still need to pass that in I think. We could do it in Run after calling newSandbox, but mutating a sandbox outside of its constructor feels a bit poor to me.

Reply:
This was the same either way since they were hidden parameters that external implementations couldn't use, but you're right, they shouldn't be there I'll just pass them in.

Reply:
Done. Waiting on CI to make sure there's not something weird in the new cleanup logic.